vowpalwabbit/slim/include/model_parser.h

```diff
@@ -0,0 +1,151 @@
+#pragma once
+
+#include <memory>
+#include <cctype>
+#include <string>
+
+#include "vw_slim_return_codes.h"
+#include "hash.h"
+
+// #define MODEL_PARSER_DEBUG
+
+#ifdef MODEL_PARSER_DEBUG
+#include <iostream>
+#include <iomanip>
+#include <fstream>
+#endif
+
+namespace vw_slim {
+
+	class model_parser
+	{
+		const char* _model_begin;
+		const char* _model;
+		const char* _model_end;
+		uint32_t _checksum;
+
+	public:
+		model_parser(const char* model, size_t length);
+
+		int read(const char* field_name, size_t field_length, const char** ret);
+		
+		int skip(size_t bytes);
+
+		const char* position();
+
+		uint32_t checksum();
+
+		template<bool compute_checksum>
+		int read_string(const char* field_name, std::string& s)
+		{
+			uint32_t str_len;
+			RETURN_ON_FAIL(read<uint32_t COMMA compute_checksum>("string.len", str_len));
+#ifdef MODEL_PARSER_DEBUG
+			{
+				std::fstream log("c:\\temp\\skype.log", std::fstream::app);
```

Comment:
We should not have a windows path hardcoded as well as skype

Reply:
This does seem rather odd to have in VW.

Reply:
Fixed.

---
In reply to: [270507573](https://github.com/VowpalWabbit/vowpal_wabbit/pull/1800#discussion_r270507573) [](ancestors = 270507573)